src/Features/Core/Portable/SplitOrMergeIfStatements/Consecutive/AbstractSplitIntoConsecutiveIfStatementsCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.SplitOrMergeIfStatements
+{
+    internal abstract class AbstractSplitIntoConsecutiveIfStatementsCodeRefactoringProvider
+        : AbstractSplitIfStatementCodeRefactoringProvider
+    {
+        // Converts:
+        //    if (a || b)
+        //        Console.WriteLine();
+        //
+        // To:
+        //    if (a)
+        //        Console.WriteLine();
+        //    else if (b)
+        //        Console.WriteLine();
+
+        // Converts:
+        //    if (a || b)
+        //        return;
+        //
+        // To:
+        //    if (a)
+        //        return;
+        //    if (b)
+        //        return;
+
+        // The second case is applied if control flow quits from inside the body.
```

Comment:
Can we just not do this bit? Personally, I don't like the style -- if there's a if/else if/else if block, it means my brain can immediately determine only one block will run _without having to analyze the guts of any of the blocks_. The moment there is two ifs, it suddenly gets a lot more complicated.

Maybe it's just my personal dislike clouding it, but it seems like there's a lot of extra work and overhead to deal with this versus just always do if/else and be done with it! CanBeSeparateStatementsAsync can just go away entirely, and the true condition of the function that calls it.

Reply:
I sympathize with that a little bit because because I liked the simplicity of only generating if/else, but @jnm2 push back against that. In fact his issue is *only* asking for generating separate if statements. It was my idea to generalize that by using if/else.

I agree with him though that this would be very useful for guard statements or "bail out" / input checks at the beginning of a method. This is a very common pattern and you don't want the rest of the method to be in an else clause. Also, supporting this in the opposite refactoring "merge consecutive if statements" means we will cover more cases (i.e. it's not just a matter of a style there). In the end, I like the way this turned out because I don't think the code is too complicated.

If you have a strong disagreement could we maybe split these into separate actions?